test/store_unittest.cc

```diff
@@ -0,0 +1,226 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  Configuration config;
+  MetadataStore store;
+
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+};
+
+TEST_F(MetadataStoreTest, ResourceID) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_EQ("some_resource", store.LookupResource("id").type());
```

Comment:
[Optional] `MonitoredResource` has an equality operator defined. So you can just compare monitored resources for equality, e.g., `EXPECT_EQ(mr, store.LookupResource("id"));`.